src/controller/model/endpoint.ts

```diff
@@ -261,26 +261,37 @@ class Endpoint extends Entity {
         return null;
     }
 
-    public sendPendingRequests(): void {
-        [...this.pendingRequests].forEach(async (r) => {
-            this.pendingRequests.splice(this.pendingRequests.indexOf(r), 1);
+    public async sendPendingRequests(): Promise<void> {
+        while (this.pendingRequests.length > 0) {
+            const r = this.pendingRequests.shift();
             try {
                 const result = await r.func();
                 r.resolve(result);
             } catch (error) {
                 r.reject(error);
             }
-        });
+        }
     }
 
-    public async sendRequest<Type>(func: () => Promise<Type>, sendWhenActive: boolean): Promise<Type> {
-        if (sendWhenActive) {
-            return new Promise((resolve, reject): void =>  {
-                this.pendingRequests.push({func, resolve, reject});
-            });
-        } else {
-            return func();
+    private async sendRequest<Type>(func: () => Promise<Type>, sendWhenActive: boolean): Promise<Type> {
+        if (this.pendingRequests.length === 0) {
+            try {
+                return await func();
+            } catch(error) {
+                // TODO: Improve check for expired transaction
+                if (!sendWhenActive || error.code !== 240) {
```

Comment:
I would not rely on the error code here, they might be different across different adapters

Reply:
Should we formalize an error type, or should we just throw it in pending regardless of the error when sendWhenActive is set?

Reply:
- sendWhenActive == false -> throw error
- sendWhenActive == true -> dont throw error

Reply:
Done, and expanded the tests to cover the new cases.